Ticked infix decls

eraserhd opened the pull request:
Allows for declaring the fixity and precedence of backtick-quoted words.

Closes #1811

msmorgan commented:
The documentation should probably mention the default fixity when using backticks.

This change seems to make fixity overloading more important. I've been thinking about some approaches for it. I'll make an issue for it on Monday if there's not one already.

Melvar commented:
a) Do backticked names with no declared fixity continue to have their own, otherwise inaccessible precedence below all user-defined precedences?
b) The changelog uses incorrect syntax. You cannot escape a backtick inside an inline code span, instead you must make the inline code span using double-backticks and leave the backticks inside the span untouched.

eraserhd commented:
@msmorgan I added a note about default precedence for backticked operators in `CHANGELOG.md`

Custom operators aren't mentioned anywhere in the reference and once in the tutorial, and it feels like it doesn't really fit in the syntax guide (I think it's trying to give an overview, not details).  Where else would it go?

@Melvar a) Yes.  b) I fixed the changelog syntax.

ahmadsalim commented:
Thanks!
